### DISCUSSION TITLE
https://starter.obytes.com/getting-started/create-new-app/

### DISCUSSION THREAD

**Discussion by giscus** [obytes/react-native-template-obytes#252] [Docs Comments]
# https://starter.obytes.com/getting-started/create-new-app/

Let's create a new React native project with obytes starter.

https://starter.obytes.com/getting-started/create-new-app/

**Comment by itsanishjain** [comment]
```


> Configure project :react-native-reanimated
Android gradle plugin: 8.6.0
Gradle: 8.10.2

> Task :app:configureCMakeDebug[arm64-v8a] FAILED
C/C++: CMake Error at C:/Users/hello/Documents/ship/umax/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake:42 (add_library):
C/C++:   Syntax error in cmake code when parsing string
C/C++:     C:\Users\hello\Documents\ship\umax\node_modules\react-native\ReactAndroid/cmake-utils/default-app-setup/OnLoad.cpp
C/C++:   Invalid character escape '\U'.
C/C++: Call Stack (most recent call first):
C/C++:   CMakeLists.txt:31 (include)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:configureCMakeDebug[arm64-v8a]'.
> [CXX1429] error when building with cmake using C:\Users\hello\Documents\ship\umax\node_modules\react-native\ReactAndroid\cmake-utils\default-app-setup\CMakeLists.txt: -- The C compiler identification is Clang 17.0.2
  -- The CXX compiler identification is Clang 17.0.2
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: C:/Users/hello/AppData/Local/Android/Sdk/ndk/26.1.10909125/toolchains/llvm/prebuilt/windows-x86_64/bin/clang.exe - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: C:/Users/hello/AppData/Local/Android/Sdk/ndk/26.1.10909125/toolchains/llvm/prebuilt/windows-x86_64/bin/clang++.exe - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Configuring incomplete, errors occurred!
  See also "C:/Users/hello/Documents/ship/umax/android/app/.cxx/Debug/3j3t62m3/arm64-v8a/CMakeFiles/CMakeOutput.log".

```

Getting this ^ error when I run pnpm android

```
jave version

java --version
openjdk 17.0.10 2024-01-16 LTS
OpenJDK Runtime Environment Microsoft-8902769 (build 17.0.10+7-LTS)
OpenJDK 64-Bit Server VM Microsoft-8902769 (build 17.0.10+7-LTS, mixed mode, sharing)

```


```
Gradle version

 gradle -v

------------------------------------------------------------
Gradle 8.11.1
------------------------------------------------------------

Build time:    2024-11-20 16:56:46 UTC
Revision:      481cb05a490e0ef9f8620f7873b83bd8a72e7c39

Kotlin:        2.0.20
Groovy:        3.0.22
Ant:           Apache Ant(TM) version 1.10.14 compiled on August 16 2023
Launcher JVM:  17.0.10 (Microsoft 17.0.10+7-LTS)
Daemon JVM:    C:\Program Files\Microsoft\jdk-17.0.10.7-hotspot (no JDK specified, using current Java home)
OS:            Windows 11 10.0 amd64


```

Any Idea why I am getting this error and where to start to fix this?

**Reply by Aarjav638** [reply]
Same Issue 

While building the app 

tried many things such as clearing caches, removing the node_modules , Gradle clean and updating the Cmake and NDK version.
also tried to update or downgrade the gradle version and also wiped out the emulator
but nothing works for me.

Please provide Idea or solution to fix this issue.

**Reply by itsanishjain** [reply]
I tried to solve it wasted 2hrs

Probably skill issue

move to expo docs

install the simple expo template code

learn about expo routes and new routing system and what build development means

now I am expert

moral of the story checkout expo doc and learn it by asking bunch of question to AI

**Comment by stanfis111** [comment]
Hello! 
Android builds via EAS are failing with the following errors. Please post fix for this otherwise awesome kit. Many thanks!

:

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.

-----------

* Where:

Build file '/home/expo/workingdir/build/node_modules/expo-updates/android/build.gradle' line: 24

* What went wrong:

A problem occurred evaluating project ':expo-updates'.

> Could not find method applyKspJvmToolchain() for arguments [] on project ':expo-updates' of type org.gradle.api.Project.

* Try:

> Run with --stacktrace option to get the stack trace.

> Run with --info or --debug option to get more log output.

> Run with --scan to get full insights.

> Get more help at https://help.gradle.org.

==============================================================================

2: Task failed with an exception.

-----------

* Where:

Script '/home/expo/workingdir/build/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle' line: 86

* What went wrong:

A problem occurred configuring project ':expo'.

> Could not get unknown property 'release' for SoftwareComponent container of type org.gradle.api.internal.component.DefaultSoftwareComponentContainer.

* Try:

> Run with --stacktrace option to get the stack trace.

> Run with --info or --debug option to get more log output.

> Run with --scan to get full insights.

> Get more help at https://help.gradle.org.

**Comment by owenhope** [comment]
I am trying to run the starter on iOS and I keep getting the following after running

pnpm ios

 (NOBRIDGE) ERROR  ReanimatedError: [Reanimated] Mismatch between JavaScript code version and Reanimated Babel plugin version (3.16.1 vs. 3.16.6).        
See `https://docs.swmansion.com/react-native-reanimated/docs/guides/troubleshooting#mismatch-between-javascript-code-version-and-reanimated-babel-plugin-version` for more details.
Offending code was: `function reactNativeReanimated_threadsTs6(){const{worklet,args,makeShareableCloneOnUIRecursive}=this.__closure;const result=worklet(...args);return makeShareableCloneOnUIRecursive(result);}` [Component Stack]
 (NOBRIDGE) WARN  Route "./(app)/_layout.tsx" is missing the required default export. Ensure a React component is exported as default. [Component Stack

I've tried to clear cache, rebuild nearly everything...any insight?

**Comment by tianshuangming** [comment]
C:\Users\24186\Desktop\rn\MyApp\android\gradlew.bat app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8087 -PreactNativeArchitectures=arm64-v8a,armeabi-v7a exited with non-zero code: 9009

**Comment by Nickcarypearson** [comment]
Error when running pnpm android:

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-reanimated:buildCMakeDebug[arm64-v8a]'.
> com.android.ide.common.process.ProcessException: ninja: Entering directory `C:\Users\nickc\WebstormProjects\jamfit-mobile\JamFit\node_modules\react-native-reanimated\android\.cxx\Debug\2p3h5q3v\arm64-v8a'
  [0/2] Re-checking globbed directories...
  ninja: build stopped: .

  C++ build system [build] failed while executing:
      @echo off
      "C:\\Users\\nickc\\AppData\\Local\\Android\\Sdk\\cmake\\3.22.1\\bin\\ninja.exe" ^
        -C ^
        "C:\\Users\\nickc\\WebstormProjects\\jamfit-mobile\\JamFit\\node_modules\\react-native-reanimated\\android\\.cxx\\Debug\\2p3h5q3v\\arm64-v8a" ^
        reanimated ^
        worklets
    from C:\Users\nickc\WebstormProjects\jamfit-mobile\JamFit\node_modules\react-native-reanimated\android
  ninja: error: mkdir(src/main/cpp/reanimated/CMakeFiles/reanimated.dir/C_/Users/nickc/WebstormProjects/jamfit-mobile/JamFit/node_modules/react-native-reanimated): No such file or directory

**Comment by OgDev-01** [comment]
Are we using the bare workflow in this version?

**Comment by MikBin** [comment]
hi there I've just create a new app following this guide, but when i run pnpm android I get the following error:
AssertionError [ERR_ASSERTION]: Invalid format of Android package name. Only alphanumeric characters, '.' and '_' are allowed, and each '.' must be followed by a letter. Reserved Java keywords are not allowed.

I've tried to create an app using proper naming convention com.example.myapp but the error persists.
any idea?
I'm on windows 11 and I don't have any android development environment installed

**Reply by assainov** [reply]
Edit `env.js` variables. Some variables are autogenerated based on your folder name. Ensure they do not contain special characters apart from ".", "_" and "-".

**Reply by MikBin** [reply]
great! it worked! thanks a lot!

**Comment by wesleysiqueira** [comment]
![image](https://github.com/user-attachments/assets/8301ef20-8999-4c9e-af84-4d5ffc9a4210)

Tried a lot to fix! But i couldn't!

**Reply by assainov** [reply]
Make sure you use x code 16.2 - the repo doesn't work with later versions. Also, make sure you reinstall simulator for the newly added x code version.

**Comment by satorusvessel** [comment]
I am getting the following error when I try to run pnpm andorid :

Failed to resolve the Android SDK path. Default install location not found: C:\Users\HP\AppData\Local\Android\Sdk. Use ANDROID_HOME to set the Android SDK location.
Failed to resolve the Android SDK path. Default install location not found: C:\Users\HP\AppData\Local\Android\Sdk. Use ANDROID_HOME to set the Android SDK location.
Error: 'adb' is not recognized as an internal or external command,
operable program or batch file.
Error: 'adb' is not recognized as an internal or external command,
operable program or batch file.
    at notFoundError (C:\Users\HP\Documents\Projects\Github\MyApp\node_modules\cross-spawn\lib\enoent.js:6:26)
    at verifyENOENT (C:\Users\HP\Documents\Projects\Github\MyApp\node_modules\cross-spawn\lib\enoent.js:40:16)
    at ChildProcess.cp.emit (C:\Users\HP\Documents\Projects\Github\MyApp\node_modules\cross-spawn\lib\enoent.js:27:25)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)
 ELIFECYCLE  Command failed with exit code 1.
